src/Components/Product/ProductListing.jsx

```diff
@@ -21,17 +33,38 @@ export const ProductListing = () => {
     inStock,
     fastDelivery,
     priceRange,
-    searchValue
+    searchValue,
+    brandFilter,
+    categoryFilter
   );
 
+  useEffect(() => {
+    (async () => {
+      try {
+        setShowLoader(true);
+        const {
+          data: { products },
+        } = await axios.get("https://api-supminn.herokuapp.com/products");
+        dispatch({ type: "SET_PRODUCTS", payload: products });
+        setShowLoader(false);
+      } catch (err) {
+        console.error(err);
+      }
+    })();
+  }, [dispatch]);
+
   useEffect(() => {
     document.title = "SupMart | Products";
     if (window.innerWidth >= 768) {
```

Comment:
if user resize their this will not work

Reply:
What is the optimal solution to be used here?

Reply:
useEffect(() => {
    updateDimensions();
    window.addEventListener("resize", updateDimensions);
    return () => window.removeEventListener("resize", updateDimensions);
  }, []);